tinygrad/codegen/linearizer.py

```diff
@@ -245,150 +374,25 @@ def render_loop(xx:List[Variable]) -> Tuple[UOp, ...]:
       self.loop_uops.update({x.expr:self.uops.add(UOps.SPECIAL, dtypes.int32, (), (len(loop_global_idxs)-1-i, x.expr, x.max+1)) for i,x in enumerate(loop_global_idxs)})  # noqa: E501
       self.loop_uops.update({x.expr:self.uops.add(UOps.SPECIAL, dtypes.int32, (), (len(loop_local_idxs)-1-i, x.expr, x.max+1)) for i,x in enumerate(loop_local_idxs)})  # noqa: E501
     else:
-      render_loop(loop_global_idxs+loop_local_idxs)
+      self.render_loop(loop_global_idxs+loop_local_idxs)
 
     # parse AST
-    loaded_buffers = {}
+    self.loaded_buffers = {}
```

Comment:
why change this?

Reply:
it needed to be accessed while rendering the reduceops and it seemed like just as much a property of the kernel as much as `load_cache`

Reply:
can it be passed into the function directly? ideally we want to keep object state minimal. load_cache is cleared after use too

Reply:
bet